pkg/cluster/cluster.go

```diff
@@ -460,7 +463,38 @@ func (c *Cluster) Delete() error {
 }
 
 func (c *Cluster) ReceivePodEvent(event spec.PodEvent) {
-	c.podEvents <- event
+	c.podEventsQueue.Add(event)
+}
+
+func (c *Cluster) podEventProcess(obj interface{}) error {
```

Comment:
I'd rename it to processPodEvent (action -> object, i.e. setStatus), or name it podEventProcessor.